soc/wch/ch32v00x/soc.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024 Michael Hope
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <ch32v003fun.h>
```

Comment:
I could see that <ch32v003fun.h> is really fun : ). In fact, it defines for the whole ch32v and ch32x series. I'll propose that you rename this file to something like "ch32vx_fun.h". It seems that the series that you are adding is ch32v nad the ch32v003 is just a variant inside that family. Is my assumption is true you need to rework SoC folder to accommodate all this.

Reply:
I see that the folder here has been renamed, the file, since it's copied verbatim from an upstream repo I think ther's some marginal value in keeping the original name, how about renaming it as a followup if it diverges? not a big deal to change it later anyway

Reply:
Done now - thanks!